packages/core/src/amazonqFeatureDev/util/fileExtension.ts

```diff
@@ -1,208 +0,0 @@
- */
-
-/**
- * Most common programming langauges and their file extensions
- * From: https://survey.stackoverflow.co/2023/#technology
- */
-export const fileExtensionAllowList = [
-    '.abap',
-    '.ada',
-    '.ads',
-    '.adb',
-    '.cls',
-    '.trigger',
-    '.apl',
-    '.asm',
-    '.s',
-    '.awk',
-    '.sh',
-    '.bash',
-    '.boo',
-    '.c',
-    '.h',
-    '.cs',
-    '.csx',
-    '.cpp',
-    '.cc',
-    '.cxx',
-    '.c',
-    '.h',
-    '.hpp',
-    '.clj',
-    '.cljc',
-    '.cljs',
-    '.cob',
-    '.cbl',
-    '.cpy',
-    '.cobra',
-    '.coffee',
-    '.cfm',
-    '.cfc',
-    '.lisp',
-    '.lsp',
-    '.v',
-    '.cr',
-    '.d',
-    '.dart',
-    '.pas',
-    '.dpr',
-    '.dfm',
-    '.e',
-    '.ex',
-    '.exs',
-    '.elm',
-    '.el',
-    '.elc',
-    '.erl',
-    '.hrl',
-    '.fs',
-    '.fsx',
-    '.fsi',
-    '.flow',
-    '.f',
-    '.f90',
-    '.f95',
-    '.gd',
-    '.go',
-    '.groovy',
-    '.gvy',
-    '.hack',
-    '.hh',
-    '.hs',
-    '.lhs',
-    '.html',
-    '.htm',
-    '.css',
-    '.hy',
-    '.pro',
-    '.idl',
-    '.io',
-    '.java',
-    '.class',
-    '.jar',
-    '.js',
-    '.mjs',
-    '.cjs',
-    '.jl',
-    '.kt',
-    '.kts',
-    '.lean',
-    '.b',
-    '.xc',
-    '.x',
-    '.xi',
-    '.lgt',
-    '.logtalk',
-    '.lua',
-    '.mpl',
-    '.mv',
-    '.m',
-    '.nb',
-    '.ma',
-    '.m',
-    '.m',
-    '.mu',
-    '.n',
-    '.nim',
-    '.nix',
-    '.m',
-    '.h',
-    '.ml',
-    '.mli',
-    '.oz',
-    '.pir',
-    '.pasm',
-    '.pas',
-    '.pp',
-    '.pl',
-    '.pm',
-    '.perl',
-    '.php',
-    '.phtml',
-    '.pike',
-    '.pmod',
-    '.ps1',
-    '.psm1',
-    '.psd1',
-    '.pro',
-    '.pl',
-    '.prolog',
-    '.pp',
-    '.purs',
-    '.py',
-    '.pyc',
-    '.pyo',
-    '.pyw',
-    '.qs',
-    '.r',
-    '.rmd',
-    '.rdata',
-    '.rkt',
-    '.rktl',
-    '.raku',
-    '.rakumod',
-    '.rakutest',
-    '.re',
-    '.res',
-    '.red',
-    '.reds',
-    '.rexx',
-    '.rex',
-    '.ring',
-    '.roff',
-    '.ron',
-    '.rm',
-    '.ms',
-    '.rb',
-    '.ruby',
-    '.rbw',
-    '.rs',
-    '.rlib',
-    '.sas',
-    '.scala',
-    '.sc',
-    '.scm',
-    '.ss',
-    '.sb3',
-    '.shen',
-    '.st',
-    '.sol',
-    '.sql',
-    '.sml',
-    '.sig',
-    '.scd',
-    '.sc',
-    '.swift',
-    '.tcl',
-    '.t',
-    '.tu',
-    '.oot',
-    '.ts',
-    '.tsx',
-    '.vala',
-    '.vapi',
-    '.vba',
-    '.bas',
-    '.v',
-    '.sv',
-    '.vhd',
-    '.vhdl',
-    '.vb',
-    '.vbx',
-    '.zig',
-    // End of Stack Overflow list, add other relevant file extensions as needed starting next line
-    '.json',
-    '.md',
-    '.scss',
-]
-
-/**
- * Validate file extension against allow list
- * @param filePath
- * @returns flag if the file is relevant to code context
- */
-export const isCodeRelatedFile = (filePath: string): boolean =>
-    fileExtensionAllowList.some((extension) => filePath.endsWith(extension))
```

Comment:
featureDev team: I moved this to `packages/core/src/shared/filetypes.ts`. I also made it case-insensitive.